trace/trace.go

```diff
@@ -246,21 +240,14 @@ func startSpanInternal(name string, hasParent bool, parent SpanContext, remotePa
 	return span
 }
 
-// EndSpan ends the current span.
-//
-// The context passed to EndSpan will still refer to the now-ended span, so any
-// code that adds more information to it, like SetSpanStatus, should be called
-// before EndSpan.
-func EndSpan(ctx context.Context) {
-	s, ok := ctx.Value(contextKey{}).(*Span)
-	if !ok {
-		return
-	}
-	s.End()
+// EndOptions represents the options
+// when ending a span.
+type EndOptions struct {
+	Tags *tag.Map
```

Comment:
I don't think we should do this for now. Recall @bogdandrutu's concern about repeated tags because they were propagated from the caller. What we really want here is to only add tags that were created in the scope of *this* span, not propagated tags (since they should be recorded in the parent Span).

Reply:
I think we should tackle this separately.